plugins/inputs/win_wmi/win_wmi.go

```diff
@@ -0,0 +1,249 @@
+var sampleConfig string
+
+// Query struct
+type Query struct {
+	query                string
+	Namespace            string   `toml:"namespace"`
+	ClassName            string   `toml:"class_name"`
+	Properties           []string `toml:"properties"`
+	Filter               string   `toml:"filter"`
+	TagPropertiesInclude []string `toml:"tag_properties"`
+	tagFilter            filter.Filter
+}
+
+// Wmi struct
+type Wmi struct {
+	Queries []Query `toml:"query"`
+	Log     telegraf.Logger
+}
+
+// S_FALSE is returned by CoInitializeEx if it was already called on this thread.
+const sFalse = 0x00000001
+
+func oleInt64(item *ole.IDispatch, prop string) (int64, error) {
+	v, err := oleutil.GetProperty(item, prop)
+	if err != nil {
+		return 0, err
+	}
+	defer func() { _ = v.Clear() }()
+
+	i := v.Val
+	return i, nil
+}
+
+// Init function
+func (s *Wmi) Init() error {
+	if err := compileInputs(s); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SampleConfig function
+func (s *Wmi) SampleConfig() string {
+	return sampleConfig
+}
+
+func compileInputs(s *Wmi) error {
+	buildWqlStatements(s)
+	if err := compileTagFilters(s); err != nil {
+		return err
+	}
+	return nil
+}
+
+func compileTagFilters(s *Wmi) error {
+	var err error
+	for i, q := range s.Queries {
+		s.Queries[i].tagFilter, err = compileTagFilter(q)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func compileTagFilter(q Query) (filter.Filter, error) {
+	tagFilter, err := filter.NewIncludeExcludeFilterDefaults(q.TagPropertiesInclude, nil, false, false)
+	if err != nil {
+		return nil, fmt.Errorf("creating tag filter failed: %v", err)
+	}
+	return tagFilter, nil
+}
+
+func buildWqlStatements(s *Wmi) {
+	for i, q := range s.Queries {
+		s.Queries[i].query = buildWqlStatement(q)
+	}
+}
+
+// build a WMI query from input configuration
+func buildWqlStatement(q Query) string {
+	wql := fmt.Sprintf("SELECT %s FROM %s", strings.Join(q.Properties, ", "), q.ClassName)
+	if len(q.Filter) > 0 {
+		wql = fmt.Sprintf("%s WHERE %s", wql, q.Filter)
+	}
+	return wql
+}
+
+func (q *Query) doQuery(acc telegraf.Accumulator) error {
+	// The only way to run WMI queries in parallel while being thread-safe is to
+	// ensure the CoInitialize[Ex]() call is bound to its current OS thread.
+	// Otherwise, attempting to initialize and run parallel queries across
+	// goroutines will result in protected memory errors.
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	tags := map[string]string{}
+	fields := map[string]interface{}{}
+
+	// init COM
+	if err := ole.CoInitializeEx(0, ole.COINIT_MULTITHREADED); err != nil {
+		oleCode := err.(*ole.OleError).Code()
+		if oleCode != ole.S_OK && oleCode != sFalse {
+			return err
+		}
+	}
+	defer ole.CoUninitialize()
+
+	unknown, err := oleutil.CreateObject("WbemScripting.SWbemLocator")
+	if err != nil {
+		return err
+	} else if unknown == nil {
+		return fmt.Errorf("failed to create WbemScripting.SWbemLocator, is WMI broken?")
+	}
```

Comment:
No need for an `else` here as we will exit in case of an error. Furthermore, in Golang errors should start with a lower-case letter and **do not contain punctuation**.
```suggestion
	if err != nil {
		return err
	} 
	if unknown == nil {
		return errors.New("failed to create WbemScripting.SWbemLocator, maybe WMI is broken")
	}
```

Reply:
Done. Thanks!